components/dashboard/src/app/AppBlockingFlows.tsx

```diff
@@ -53,5 +56,9 @@ export const AppBlockingFlows: FC = ({ children }) => {
         return <UserOnboarding user={user} />;
     }
 
+    if (!org.data) {
+        return <OrgNamingStep onComplete={() => {}} />;
```

Comment:
**praise**: Tested also with users joining a single organization and leaving that organization—worked like a charm.